src/engine/threaded_engine_perdevice.cc

```diff
@@ -113,8 +113,8 @@ class ThreadedEnginePerDevice : public ThreadedEngine {
         if (is_copy) {
           auto ptr =
           gpu_copy_workers_.Get(ctx.dev_id, [this, ctx, is_copy, nthread]() {
-            // Signify to kernel that GPU is being used,  no Kernel Launch OMP (temporary behavior)
-            OpenMP::Get()->set_enabled(false);
+            // Signify to kernel that GPU is being used, so reserve cores as necessary
+            OpenMP::Get()->set_reserve_cores(GetReserveCoreCount(true));
```

Comment:
Is it necessary to reserve at lease 1 even when not using GPU? For python frontend thread?

Reply:
we are already at physical cores / 2, but I can save another physical core if you like.  Just let me know.

Reply:
How about we add one for CPU if # real cores >= 4?

Reply:
Again, if you know of a workflow that may behave badly, I am happy to run tests with it.  I only play with a few models, like MNIST, CIFAR, DI LSTM (I didn't run the LSTM one for these tests because they tend to run fastest for OMP_NUM_THREADS=1)

Reply:
Also note that since adding gperftools/jemalloc, I have never seen 100% CPU utilization -- even on a 4-core desktop.  Usually it's under 50%.

Reply:
ok